warehouse/warehouse.go

```diff
@@ -797,10 +861,21 @@ func (wh *HandleT) getUploadsToProcess(availableWorkers int, skipIdentifiers []s
 		lastStatus, upload.LastAttemptAt = warehouseutils.TimingFromJSONString(lastTiming)
 		upload.Attempts = gjson.Get(string(upload.Error), fmt.Sprintf(`%s.attempt`, lastStatus)).Int()
 
+		if upload.WorkspaceID == "" {
+			var ok bool
+			wh.workspaceBySourceIDsLock.Lock()
+			upload.WorkspaceID, ok = wh.workspaceBySourceIDs[upload.SourceID]
+			wh.workspaceBySourceIDsLock.Unlock()
+
+			if !ok {
+				pkgLogger.Warnf("could not find workspace id for source id: %s", upload.SourceID)
```

Comment:
What might cause the code to enter into this loop? Can we have a stat for this miss of workspaceID?